src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs

```diff
@@ -562,10 +560,9 @@ private void ApplyInsets(Thickness insets)
 				{
 					return;
 				}
-
-				if (_insetMode == InsetMode.Padding
-					&& !PaddingHelper.GetPadding(owner).Equals(insets)
-					&& PaddingHelper.SetPadding(owner, insets))
+				if (_insetMode == InsetMode.Padding &&
+					!PaddingHelper.GetPadding(owner).Equals(insets) &&
+					PaddingHelper.SetPadding(owner, insets))
```

Comment:
you can create a new method `TryUpdatePadding` (TrySet was taken) that checks for inequality and then assigns value
currently we are doing reflection(even if memorized) on the property twice shortly when that could be avoided